v2/client.go

```diff
@@ -73,6 +77,14 @@ func NewClient(config *ClientConfiguration) (Client, error) {
 		c.AuthConfig = config.AuthConfig
 	}
 
+	if config.EnableAlphaFeatures && config.OriginatingIdentity != nil {
```

Comment:
hmmm This implies we need to create a new client each time we want to switch users.... I can't decide if that's better than to just allow people to specify it on the request struct itself.  I'm imagining a case where we have one client object on the platform and they want to reuse it all requests to a broker - might be a bit much to ask them to create one per user as well.